### DISCUSSION TITLE
File upload to all hosts

### DISCUSSION THREAD

**Discussion by infinite1by0** [ansibleguy76/ansibleforms#137] [Q&A] [answered]
If i create a file upload field in form, how can i access the file in awx playbook. Actually I want to copy the file to all hosts

**Comment by hmarko75** [comment]
Cant you use nfs mount from all hosts to the upload folder?

**Reply by ansibleguy76** [reply]
mounting would be one way... but a loop copy would be easiest I guess (see answer below)

**Comment by ansibleguy76** [comment, marked as answer]
The files get uploaded on the AnsibleForms host with temp names and extravars are sent with info about location and filenames.  
The best way would be to create a playbook on your AWX that invokes a loop copy to all your hosts.  
If you create a keypair on the AWX host 
`ssh-keygen -t rsa`
And then copy the public key to authorized_keys on all hosts
`scp ~/.ssh/id_rsa.pub root@172.16.50.1:~/.ssh/authorized_keys`
you should be able to scp passwordless.  Then simple loop and copy and rename the file to original name for each host.
